linux_sql/scripts/host_info.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# Script usage
+#./scripts/host_info.sh psql_host psql_port db_name psql_user psql_password
+# Example
+#./scripts/host_info.sh "localhost" 5432 "host_agent" "postgres" "mypassword"
+
+#error function
+errorf(){
+  echo $1 1>&2
+  exit 1
+}
+
+#assign CLI arguments to variables (e.g. `psql_host=$1`)
+psql_host=$1
+psql_port=$2
+db_name=$3
+psql_user=$4
+psql_password=$5
+
+#refrain from needing to manually enter password
+export PGPASSWORD=$psql_password
+
+#save hostname to a variable
+hostname="$(hostname -f)"
+
+#host hardware specs raw data
+lscpu_out=$(lscpu)
+meminfo_out=$(cat /proc/meminfo)
+
+#parse host hardware specifications using bash cmds
+cpu_number=$(echo "$lscpu_out"  | grep -E "^CPU\(s\):" | awk '{print $2}' | xargs)
+if [ "$cpu_number" = "" ]; then
+  errorf "failed to get cpu_number"
```

Comment:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself
You could make a function to check empty variables